### PR TITLE
Prevent MaxListenersExceededWarning in PassThrough

### PR DESCRIPTION
**Pull request by cjh980402**
This is the solution of https://github.com/play-dl/play-dl/issues/52

### CODE REVIEW

**Comment by killer069** [comment]
@cjh980402 

Maybe this is not a good idea for Live Streams.

Normal streams are ok. But Live streams, I still doubt.

Check if this runs well for Live Streams or not.

**Comment by cjh980402** [comment]
It works well, but In live stream it was useless code. However, I found more better code to Normal Stream.

**Comment by killer069** [comment]
I need to do some testing before merging it.

File: play-dl/YouTube/classes/LiveStream.ts [hunk 1]
```diff
@@ -172,6 +170,7 @@ export class Stream {
             this.bytes_count += chunk.length
             if(absolute_bytes > (this.per_sec_bytes * 300) && this.per_sec_bytes !== 0){
                 stream.destroy()
+                stream.unpipe(this.stream)
```
Comment (killer069) [inline play-dl/YouTube/classes/LiveStream.ts:173]: Remove this line

Add this same line with this.request at setTimeout function.

That would be better.

**Review by killer069 — CHANGES_REQUESTED** [review]

File: play-dl/YouTube/classes/LiveStream.ts (hunk 1 above)
Comment (killer069) [reply play-dl/YouTube/classes/LiveStream.ts:173]: Since this will make un-piping more fast before the actual data is completed.

**Review by killer069 — APPROVED** [review]

**Comment by killer069** [comment]
will add this in few hours.

**Comment by cjh980402** [comment]
Is `this.request?.unpipe(this.stream)` is better than `stream.unpipe(this.stream)`  in that code?

**Comment by killer069** [comment]
> Is `this.request?.unpipe(this.stream)` is better than `stream.unpipe(this.stream)` in that code?

Yes, since if some data is let over for buffering and un-piping code forcingly may cause errors in streams.

**Pull request merged**